### PR TITLE
Add full placeholder when it's convenient

### PR DESCRIPTION
**Pull request by FoggyFinder**
Close: #47

### CODE REVIEW

File: CSharpMath.Editor/Keyboards/MathKeyboard.cs [hunk 1]
```diff
@@ -97,20 +97,45 @@ public MathListIndex InsertionIndex
...
+        static bool IsFullPlaceholderRequired(IMathAtom mathAtom) {
+          var atomType = mathAtom.AtomType;
+          switch (atomType) {
+            case MathAtomType.Variable:
+            case MathAtomType.Number:
+            case MathAtomType.Radical:
+            case MathAtomType.Fraction:
+            case MathAtomType.Ordinary:
+            // functions are treated as a LargeOperator so no special case here. Yet?
+            case MathAtomType.LargeOperator:
```
Comment (Happypig375) [inline CSharpMath.Editor/Keyboards/MathKeyboard.cs]: `LargeOperators` have special handling for superscripts and subscripts. They always support them.

File: CSharpMath.Editor/Keyboards/MathKeyboard.cs [hunk 2]
```diff
@@ -97,20 +97,45 @@ public MathListIndex InsertionIndex
...
+            case MathAtomType.Close:
+              return false;
+            default:
+              return true;
+          }
```
Comment (Happypig375) [inline CSharpMath.Editor/Keyboards/MathKeyboard.cs]: Will a blacklist-based approach be better? It seems that adding a placeholder is the minority case and new cases for `MathAtomType` should support direct scripts by default.

Comment (FoggyFinder) [reply CSharpMath.Editor/Keyboards/MathKeyboard.cs]: Maybe even black & white lists with default false?

Comment (Happypig375) [reply CSharpMath.Editor/Keyboards/MathKeyboard.cs]: In this case, a switch expression should be used for exhaustiveness warnings.

Comment (FoggyFinder) [reply CSharpMath.Editor/Keyboards/MathKeyboard.cs]: Played a bit with both approaches. Now I also see more advantages of having the single list. You're totally right.

**Pull request merged**